packages/components/src/ComponentUtils.ts

```diff
@@ -0,0 +1,80 @@
+import React, {
+  ComponentType,
+  ForwardRefExoticComponent,
+  RefAttributes,
+} from 'react';
+import { ForwardRef } from 'react-is';
+
+export type Props = Record<string, unknown> | RefAttributes<unknown>;
+
+/**
+ * Type that represents a component that has been wrapped by another component.
+ */
+export type WrappedComponentType<
```

Comment:
I was debating change the `XComponentType` to use `WrappedComponent` instead of `Original`, but unsure I actually want them to be the same property so I kept them different.